lib/matplotlib/docstring.py

```diff
@@ -72,9 +72,4 @@ def do_copy(target):
 # Create a decorator that will house the various docstring snippets reused
 # throughout Matplotlib.
 interpd = Substitution()
-
-
-def dedent_interpd(func):
-    """Dedent *func*'s docstring, then interpolate it with ``interpd``."""
-    func.__doc__ = inspect.getdoc(func)
```

Comment:
Changing `getdoc` to `cleandoc` doesn't have an effect?

Reply:
one is called on the function, the other on the docstring?

Reply:
`getdoc` says: "If the documentation string for an object is not provided and the object is a class, a method, a property or a descriptor, retrieve the documentation string from the inheritance hierarchy."

This inheritance doesn't apply to `func.__doc__`, no?

Reply:
Ah indeed.  On the other hand this was accidentally changed in the other direction in https://github.com/matplotlib/matplotlib/pull/13125/files#diff-0b973b83a5020c0418842e20e3ffc56cR115 so I guess I can always argue it's just going back to the old behavior :)

Reply:
Something broke doc build though, though I'm not sure if it was this change specifically.

Reply:
oops, had a fix but forgot to push it

Reply:
AFAICT, I could not find any `dedent_interpd` on things that didn't have their own docstring.